scarb-metadata/src/lib.rs

```diff
@@ -106,6 +106,30 @@ impl fmt::Display for CompilationUnitId {
     }
 }
 
+/// An "opaque" identifier for a compilation unit component.
+/// It is possible to inspect the `repr` field if the need arises,
+/// but its precise format is an implementation detail and is subject to change.
+///
+/// [`Metadata`] can be indexed by [`CompilationUnitComponentId`].
```

Comment:
you haven't added that indexing operation

Reply:
Added indexing to `CompilationUnitMetadata`